app/commands.py

```diff
@@ -248,29 +249,38 @@ def backfill_processing_time(start_date, end_date):
         send_processing_time_for_start_and_end(process_start_date, process_end_date)
 
 
-@notify_command()
-def populate_annual_billing():
+@notify_command(name='latest_annual_billing')
+@click.option('-y', '--year', required=True,
+              help="""The year to populate the annual billing data for, i.e. 2019""")
+def populate_annual_billing(year):
     """
-    add annual_billing for 2016, 2017 and 2018.
+    add annual_billing for given year.
     """
-    financial_year = [2016, 2017, 2018]
-
-    for fy in financial_year:
-        populate_data = """
-        INSERT INTO annual_billing(id, service_id, free_sms_fragment_limit, financial_year_start,
-                created_at, updated_at)
-            SELECT uuid_in(md5(random()::text || now()::text)::cstring), id, 250000, {}, '{}', '{}'
-            FROM services
-            WHERE id NOT IN(
-                SELECT service_id
-                FROM annual_billing
-                WHERE financial_year_start={})
-        """.format(fy, datetime.utcnow(), datetime.utcnow(), fy)
-
-        services_result1 = db.session.execute(populate_data)
-        db.session.commit()
+    sql = """
+        Select id from services where active = true
+        except
+        select service_id
+        from annual_billing
+        where financial_year_start = :year
+    """
```

Comment:
i don't really mind if this doesnt get changed, but does this do anything different to a more normal join?

```select service_id
from annual_billing
join services on service_id = services.id
where service.active and financial_year_start = :year```

Reply:
wait disregard that i see this is selecting the inverse of the join - those services that dont have annual billing. thats a bit grosser query `select s.id from services s left outer join annual_billing ab on ab.service_id = s.id and ab.financial_year_start = :year where ab.id is null` which i dont think is as nice as this.